Sources/Noora/Components/CollapsibleStep.swift

```diff
@@ -0,0 +1,130 @@
+            try await task { line in
+                standardPipelines.output.write(content: "  \(line.formatted(theme: theme, terminal: terminal))\n")
+            }
+            standardPipelines.output.write(content: "  \(formattedSuccessMessage(color: theme.secondary))\n")
+        } catch {
+            standardPipelines.output.write(content: "  \(formattedErrorMessage(color: theme.secondary))\n")
+            throw error
+        }
+    }
+
+    func runInteractive() async throws {
+        renderInteractiveLines(lines: [])
+        var lines: [TerminalText] = []
+        do {
+            try await task { line in
+                lines.append(line)
+                if lines.count > visibleLines {
+                    lines.removeFirst()
+                }
+                renderInteractiveLines(lines: lines)
+            }
+        } catch {
+            renderInteractiveError()
+            throw error
+        }
+        renderInteractiveSuccess()
+    }
+
+    private func renderInteractiveSuccess() {
+        renderer.render(
+            ProgressStep.completionMessage(formattedSuccessMessage(color: theme.primary), theme: theme, terminal: terminal),
+            standardPipeline: standardPipelines.output
+        )
+    }
+
+    private func formattedSuccessMessage(color: String) -> String {
+        if let successMessage {
+            "\(successMessage.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(color, terminal)
+                .boldIfColoredTerminal(terminal)
+        } else {
+            "\(title.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(color, terminal)
+                .boldIfColoredTerminal(terminal)
+        }
+    }
+
+    private func renderInteractiveError() {
+        renderer.render(
+            ProgressStep.errorMessage(formattedErrorMessage(color: theme.primary), theme: theme, terminal: terminal),
+            standardPipeline: standardPipelines.error
+        )
+    }
+
+    private func formattedErrorMessage(color: String) -> String {
+        if let errorMessage {
+            "\(errorMessage.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(color, terminal)
+                .boldIfColoredTerminal(terminal)
+        } else {
+            "\(title.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(color, terminal)
+                .boldIfColoredTerminal(terminal)
+        }
+    }
+
+    private func renderInteractiveLines(lines: [TerminalText]) {
+        var content = ""
+        content = "◉ \(title.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(theme.primary, terminal)
+            .boldIfColoredTerminal(terminal)
+        for (index, line) in lines.enumerated() {
+            if index == 0 {
+                content.append("\n  \(line.formatted(theme: theme, terminal: terminal))")
+            } else {
+                content.append("\n  \(line.formatted(theme: theme, terminal: terminal))")
+            }
```

Comment:
I'm failing to see how the `if-else` branches differ 🤔

Reply:
This is some leftover from an experiment that I was doing with styles.